pow/src/main/java/tech/pegasys/artemis/pow/DepositContractListener.java

```diff
@@ -61,13 +65,11 @@ public DepositContractListener(Web3j web3j, EventBus eventBus, DepositContract c
     subscriptionNewDeposit =
         contract
             .depositEventEventFlowable(depositEventFilter)
-            .flatMap(this::convertToDeposit)
-            .subscribe(eventBus::post);
-  }
-
-  private Flowable<Deposit> convertToDeposit(final DepositEventEventResponse event) {
-    return getBlockByHash(event.log.getBlockHash())
-        .map(block -> new Deposit(event, UnsignedLong.valueOf(block.getTimestamp())));
+            .flatMap(
+                event ->
+                    getBlockByHash(event.log.getBlockHash())
+                        .map(block -> Pair.of(block, new Deposit(event))))
```

Comment:
Doesn't look like we're handling reorgs where events might be removed, but I guess that's covered in the follow-up ticket we have to respect the follow distance?

Reply:
If we're ultimately going to have to buffer eth1 event sets based on the follow distance, is `PublishOnInactivityDepositHandler` going to eventually go away?

Reply:
Yeah, deliberately ignoring forks for now - I'm not sure how it will all play out yet but dealing with forks and the follow distance feel like they need to be sorted out together.

We will still need `PublishOnInactivityDepositHandler` though because it's possible, and maybe even likely, that we'd go at least 1024 blocks without a new deposit.